smarts/core/sumo_road_network.py

```diff
@@ -324,10 +324,15 @@ def oncoming_lanes_at_offset(self, offset: float) -> List[RoadMap.Lane]:
             if not nearby_lanes:
                 return result
             my_vect = self.vector_at_offset(offset)
-            my_norm = np.linalg.norm(my_vect)
+            my_norm = max(np.linalg.norm(my_vect), 1e-6)
             threshold = -0.995562  # cos(175*pi/180)
```

Comment:
This seems unclear, what is the reason for this threshold and the specific value?

Reply:
for `oncoming_lanes_at_offset` we are checking for lanes that are in opposite direction of travel to the current lane. We find the heading of all the nearby lanes to this offset and check whether the dot product of the current lane's vector and nearby lane vector is `cos(175*pi/180)` which is the above threshold to find if they are in opposite direction to the given lane or not.